Model/Checkout/PlaceOrder.php

```diff
@@ -245,6 +250,15 @@ public function createOrder(Quote $quote, string $amwalOrderId, string $refId):
             $order->setAmwalOrderId($amwalOrderId . '-canceled');
             $this->orderRepository->save($order);
         }
+        if($this->config->isQuoteOverrideEnabled()){
+            $quote->setStoreId($this->storeManager->getStore()->getId());
+            $quote->setStoreCurrencyCode($this->storeManager->getStore()->getBaseCurrencyCode());
```

Comment:
Is using the base currency values the best option here. If the currency can be dynamic, it could be that a different currency then the base currency is used. What would happen in this case?

Reply:
Currently I think Amwal is just use single currency and the button will be hidden if the currency is not SAR